src/renderer/components/data-settings/data-settings.js

```diff
@@ -901,48 +903,63 @@ export default defineComponent({
         const playlistObjectKeys = Object.keys(playlistObject)
         const playlistObjectHasAllRequiredKeys = requiredKeys.every((k) => playlistObjectKeys.includes(k))
 
-        if (playlistObjectHasAllRequiredKeys) {
-          const existingPlaylist = this.allPlaylists.find((playlist) => {
-            return playlist.playlistName === playlistObject.playlistName
-          })
+        if (!playlistObjectHasAllRequiredKeys) {
+          const message = this.$t('Settings.Data Settings.Playlist insufficient data', { playlist: playlistData.playlistName })
+          showToast(message)
+          return
+        }
 
-          if (existingPlaylist !== undefined) {
-            playlistObject.videos.forEach((video) => {
-              let videoExists = false
-              if (video.playlistItemId != null) {
-                // Find by `playlistItemId` if present
-                videoExists = existingPlaylist.videos.some((x) => {
-                  // Allow duplicate (by videoId) videos to be added
-                  return x.videoId === video.videoId && x.playlistItemId === video.playlistItemId
-                })
-              } else {
-                // Older playlist exports have no `playlistItemId` but have `timeAdded`
-                // Which might be duplicate for copied playlists with duplicate `videoId`
-                videoExists = existingPlaylist.videos.some((x) => {
-                  // Allow duplicate (by videoId) videos to be added
-                  return x.videoId === video.videoId && x.timeAdded === video.timeAdded
-                })
-              }
+        const existingPlaylist = this.allPlaylists.find((playlist) => {
+          return playlist.playlistName === playlistObject.playlistName
```

Comment:
It probably makes sense to try to match by `_id` first, because if you rename the playlist on one computer you still want it to be the same when you import it on the other computer instead of being treated as a completely separate playlist.

That being said I've just realised that the `_id` is considered an ignored key, so there is no way that a renamed playlist could ever match, because the import will make sure that the `_id` won't match, but that problem already existed before this pull request. Also weird that we don't save the lastPlayedAt field when it's a new playlist.

Reply:
Agree but no way this belongs to this PR (plus need test cases)
I can make another PR to make it find by `_id` first later

`lastPlayedAt` not saved for new/existing playlist intentionally
Obvious for existing playlist but for new playlist it's never played locally and I consider it as a "local property" (e.g. importing other people's playlist)

Reply:
I have to say I personally never use the playlists, but the way I use the other export and import options is for backup purposes, which means I only ever import my own stuff in my FreeTube install (dev mode is an exception because there I only use it to test stuff and that data is saved in a separate location), I would expect everything possible to be imported.

Importing someone else's files is probably a pretty rare use case, that is only really becoming possible because of the open PR to export single playlists.